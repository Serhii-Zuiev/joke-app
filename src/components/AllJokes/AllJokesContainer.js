import React from "react";
import {
  StyledJokesContainer,
  StyledJokesHeading,
  StyledForm,
} from "../../styled";
import { JokesList } from "./";
import { Form, Button } from "react-bootstrap";

const AllJoke = () => {
  return (
    <StyledJokesContainer>
      <StyledJokesHeading>
        Here are our best jokes, but it isn't certain...
      </StyledJokesHeading>

      <Form style={{ width: "6rem" }}>
        <Form.Group controlId="exampleForm.SelectCustom">
          <Form.Label style={{ width: "8rem" }}>choose category</Form.Label>
          <Form.Control as="select" custom>
            <option>All</option>
            <option>Nerdy</option>
            <option>Explicit</option>
          </Form.Control>
        </Form.Group>
      </Form>

      <JokesList />

      <StyledForm>
        <Form.Control
          type="number"
          min="1"
          max="20"
          placeholder="1-20"
          style={{ width: "5rem" }}
        />
        <Button variant="primary" type="submit">
          Get more
        </Button>
      </StyledForm>
    </StyledJokesContainer>
  );
};

export default AllJoke;
