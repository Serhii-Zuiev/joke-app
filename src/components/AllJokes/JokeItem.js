import React from "react";
import { NavLink } from "react-router-dom";
import { Card, Button } from "react-bootstrap";
import { Eyeglasses } from "react-bootstrap-icons";
import { PatchExclamation } from "react-bootstrap-icons";
import { EmojiSmile } from "react-bootstrap-icons";
import { StyledCardContainer } from "../../styled";

function defineJokeCategory(category) {
  let Icon;
  let iconTitle;

  switch (category) {
    case "nerdy":
      Icon = Eyeglasses;
      iconTitle = "Nerdy";
      break;
    case "explicit":
      Icon = PatchExclamation;
      iconTitle = "Explicit";
      break;
    default:
      Icon = EmojiSmile;
      iconTitle = "Funny";

  }
  return { Icon, iconTitle}
}

const JokeItem = ({ text, categories, jokeID }) => {
  
  const category = defineJokeCategory(categories[0])

  return (
    <StyledCardContainer>
      <Card style={{ height: "100%" }}>
        <Card.Body className="cardBody">
          <Card.Text style={{ marginBottom: "0" }}>
            {text.length > 60 ? `${text.slice(0, 60)}...` : text}
          </Card.Text>
          <div className="JokeCategoryWrapper">

            <NavLink  to={{ pathname: `/joke/${jokeID}`, state: { id: jokeID, from: '/' } }}>
              <Button variant="outline-primary">Read joke</Button>
            </NavLink>

            <div className="JokeCategory" title={category.iconTitle}>
              <category.Icon color="#777" size={30} />
            </div>
          </div>
        </Card.Body>
      </Card>
    </StyledCardContainer>
  );
};

export default JokeItem;
