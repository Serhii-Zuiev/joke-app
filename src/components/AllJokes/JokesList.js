import React from "react";
import { StyledJokesList } from "../../styled";
import { JokeItem } from "./";
import { placeholder } from "../../placeholder";

const JokesList = () => {
  return (
    <StyledJokesList>
      {placeholder.map((item) => (
        <li key={item.id} className="jokesListItem">
          <JokeItem text={item.joke} />
        </li>
      ))}
    </StyledJokesList>
  );
};

export default JokesList;
