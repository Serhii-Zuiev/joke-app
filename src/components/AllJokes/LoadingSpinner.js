import React from "react";
import Spinner from "react-bootstrap/Spinner";
import { StyledSpinnerContainer } from "../../styled";

const LoadingSpinner = () => (
  <StyledSpinnerContainer>
    <Spinner animation="border" variant="primary" role="status">
      <span className="sr-only">Loading...</span>
    </Spinner>
  </StyledSpinnerContainer>
);

export default LoadingSpinner;
