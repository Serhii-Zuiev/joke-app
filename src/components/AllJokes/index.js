export { default as AllJokesContainer } from "./AllJokesContainer";
export { default as JokesList } from "./JokesList";
export { default as JokeItem } from "./JokeItem";