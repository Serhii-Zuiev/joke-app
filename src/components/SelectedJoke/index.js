export { default as SelectedJoke } from "./SelectedJoke";