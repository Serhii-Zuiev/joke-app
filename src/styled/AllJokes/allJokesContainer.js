import styled from "styled-components";

export const StyledJokesContainer = styled.section`
  padding-top: 40px;

  .categorySelect {
    height: 34px;
    width: 92px;
    margin-top: 10px;
    text-align: center;
    border: 1px solid #333;
    font-size: 18px;
    color: #333;
    border-radius: 4px;
    padding: 4px;
  }
`;

export const StyledJokesHeading = styled.h2`
  font-size: 20px;
  color: #333;
  padding: 12px 0;
  letter-spacing: 0.3px;
`;

export const StyledForm = styled.form`
  width: 230px;
  margin: 0 auto;
  padding-bottom: 60px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .jokesAmmountInput {
    height: 34px;
    width: 70px;
    text-align: center;
    border: 1px solid #333;
    font-size: 18px;
    color: #333;
    border-radius: 4px;
    padding: 4px;
  }
  .loadMoreBtn {
    height: 34px;
    width: 120px;
    border: 1px solid #333;
    font-size: 18px;
    color: #333;
    border-radius: 4px;
    padding: 4px;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    &:hover {
      color: tomato;
      border: 1px solid tomato;
    }
  }
`;
