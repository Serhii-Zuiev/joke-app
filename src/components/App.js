import React from "react";
import styled from "styled-components";
import AllJokes from "./AllJokes/AllJokesContainer";
import SelectedJoke from "./SelectedJoke/SelectedJoke";

function App() {
  return (
    <StyledAppContainer>
      <StyledHeader>
        <StyledAppHeading>
          <span role="img" aria-label="face emoji">
            ðŸ˜„
          </span>
          Joke app
        </StyledAppHeading>
      </StyledHeader>

      <AllJokes />
      {/* <SelectedJoke /> */}
    </StyledAppContainer>
  );
}

export default App;

const StyledAppContainer = styled.div`
  background-color: #eee;
`;

const StyledHeader = styled.header`
  height: 70px;
  width: 90%;
  margin: 0 auto;
`;

const StyledAppHeading = styled.h1`
  font-size: 26px;
  color: #333;
  padding: 16px 0;
  letter-spacing: 1px;
  text-align: center;
  border-bottom: 2px solid #777;
`;
