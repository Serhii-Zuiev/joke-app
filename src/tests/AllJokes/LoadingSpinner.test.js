import React from "react";
import { LoadingSpinner } from "../../components/AllJokes";

describe("<LoadingSpinner/>  unit test", () => {
  it("should render LoadingSpinner component", () => {
    const component = render(<LoadingSpinner />);
    expect(component).toMatchSnapshot();
  });
});
