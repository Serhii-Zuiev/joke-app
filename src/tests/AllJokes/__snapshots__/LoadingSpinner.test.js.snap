// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner/>  unit test should render LoadingSpinner component 1`] = `
<div
  class="sc-fzoLsD kYZhHX"
>
  <div
    class="spinner-border text-primary"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
